### PR TITLE
Add all or nothing grading capabilities

### PR DESCRIPTION
**Pull request by ddxtanx**
Add action parameter to have grading dealt with in an all or nothing fashion.

### CODE REVIEW

**Review by wadefagen — CHANGES_REQUESTED** [review]
This looks great.  Only concern I have is that I think the points will still show when `all_or_nothing` is enabled?

Can you update the step summary (runner.ts:245 and :252) to have a dash (probably &mdash;, —) when `all_or_nothing` is enabled?  The total score would display correctly, but that way there's no implied partial credit at all.

**Comment by ddxtanx** [comment]
Can we have it so that the number of points appears as a dash, but we still give a checkmark/X if the students passes/fails the given testcase?

**Comment by ddxtanx** [comment]
That way a student knows what's passing/failing but still sees that they will earn no points.

**Comment by ddxtanx** [comment]
@wadefagen The latest commit should now have these capabilities!

File: src/runner.ts [hunk 1]
```diff
@@ -257,6 +259,22 @@ export const runAll = async (tests: Array<Test>, cwd: string, testSuite = 'autog
...
+      row[1] = '0'
+      return row
+    })
+
+    jsonScoreLog = jsonScoreLog.map(row => {
+      row.success = false
+      row.points = 0
+      return row
+    })
+  }
```
Comment (wadefagen) [inline src/runner.ts]: I'm a little hesitant to replace all the entire table after it was created.   

Would a simpler impl would be the following in the lines above that construct the table in the first place (for both the success and fail cases)?
```
pointsString = test.points ? test.points.toString() : '0'
if (all_or_nothing) { pointsString = "—" }
...
summaryTable.push([test.name, pointsString, '✅'])
```

**Review by wadefagen — CHANGES_REQUESTED** [review]
I agree with keeping the ✅/❌ progress -- I don't think this current change adds a dash?  See the in-code comment about a possibly impl.

File: src/runner.ts (hunk 1 above)
Comment (ddxtanx) [reply src/runner.ts]: This now appears in the main version of the code!

**Review by wadefagen — APPROVED** [review]
lgtb

**Pull request merged**